### ISSUE TITLE
Feature request: add GOBIN to PATH even if a version isn't installed

### ISSUE THREAD

**Issue by ydnar** [actions/setup-go#49]
Request: setup-go should always add GOBIN to PATH, even if it uses the default Go installed in the environment. This would mirror how it always sets up Go matchers.

Per thread here: https://github.com/actions/setup-go/issues/39#issuecomment-607420683

Thanks!

**Comment by filipesilvaa** [comment]
@ydnar I tried now with new version (setup-go@v2), and apparently it's adding GOBIN to PATH, working well at my repository.

**Comment by ydnar** [comment]
> @ydnar I tried now with new version (setup-go@v2), and apparently it's adding GOBIN to PATH, working well at my repository.

Did you specify `go-version`?

**Comment by rubenstolk** [comment]
The env var is set but it doesn't create the dir, shouldn't that happen too?

```Installation requires your GOBIN directory /home/runner/go/bin to exist. Please create it.```

**Comment by remorses** [comment]
I had the same issue, the GOBIN env variables is empty if you dont specify a go version, this is more a bug than a feature request

**Comment by subins2000** [comment]
Seems like using this one-liner `- uses: actions/setup-go@v2` just for setting the env won't work out

**Comment by bryanmacfarlane** [comment]
@subins2000 adding that one liner does add gobin to the path.

Just tried.  Look at the path step in [this run](https://github.com/bryanmacfarlane/actions-playground/runs/1374473526?check_suite_focus=true). 

The second entry of the path in this example has `/home/runner/go/bin` added

<img width="488" alt="Screen Shot 2020-11-09 at 9 05 39 AM" src="https://user-images.githubusercontent.com/919564/98551110-f0fe2f80-226a-11eb-9e09-4e3d521c2187.png">

**Comment by subins2000** [comment]
@bryanmacfarlane It didn't work for me on a Windows host and bash, so had to do :
```
echo "::set-env name=GOPATH::$(go env GOPATH)"
echo "::add-path::$(go env GOPATH)/bin"
```

**Comment by bryanmacfarlane** [comment]
I'll try it out on windows.

The code is adding it to the path (it actually does your second line of add-path - even on windows).  Note it does not set GOPATH as that's a < go 1.12 and pre go modules concept.

**Comment by leighmcculloch** [comment]
In my builds this yml is not resulting in GOBIN being in the PATH. If I go get tools they install fine but aren't available on the PATH.

This is a snippet from my workflow:
```yml
name: build
on: [push, pull_request]
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/setup-go@v2
    - uses: actions/setup-ruby@v1
    - run: go get github.com/mattn/goreman
    - run: goreman ...
```

This is what I see if I echo PATH:
```
/opt/hostedtoolcache/Ruby/2.7.2/x64/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/opt/pipx_bin:/usr/share/rust/.cargo/bin:/home/runner/.config/composer/vendor/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

**Comment by leighmcculloch** [comment]
@bryanmacfarlane I'm not sure how it works for you, because in your output there is the line `Added go to the path`, and the only place that line is printed is below and is wrapped in a `if (versionSpec)`:
https://github.com/actions/setup-go/blob/d0c5defdf364f1d1fb07530c000084836192af9c/src/main.ts#L23-L31

**Comment by leighmcculloch** [comment]
I've opened a PR #89 that adds GOBIN to PATH regardless of whether a version is specified. I tested it on my own repo and it worked in both cases.

**Comment by leighmcculloch** [comment]
@Sergey-Murtazin PR with a fix is open at #89.

**Comment by dmitry-shibanov** [comment]
Hello everyone. We added this logic in new major action version. For now I'm going to close the issue. If you have any concerns feel free to ping us.
